pkg/auth/providers/keycloakoidc/keycloak_client.go

```diff
@@ -37,15 +37,10 @@ type KeyCloakClient struct {
 	httpClient *http.Client
 }
 
-func (k *KeyCloakClient) newClient(config *v32.OIDCConfig) (KeyCloakClient, error) {
-	keyCloakClient := KeyCloakClient{}
-	if config.Certificate != "" && config.PrivateKey != "" {
-		err := oidc.GetClientWithCertKey(keyCloakClient.httpClient, config.Certificate, config.PrivateKey)
-		if err != nil {
-			return KeyCloakClient{}, err
-		}
-	}
-	return keyCloakClient, nil
+func newClient(config *v32.OIDCConfig) (*KeyCloakClient, error) {
+	keyCloakClient := &KeyCloakClient{}
+	err := oidc.GetClientWithCertKey(keyCloakClient.httpClient, config.Certificate, config.PrivateKey)
```

Comment:
nit: you could move this func into provider file

Reply:
Moved